pbpy/pbtools.py

```diff
@@ -140,94 +154,144 @@ def remove_file(file_path):
+    elif "rewinding head" in out and not ("error" in out or "conflict" in out):
+        handle_success()
+    elif "successfully rebased and updated" in out:
+        handle_success()
+    elif "failed to merge in the changes" in out or "could not apply" in out:
+        handle_error("Aborting the rebase. Changes on one of your commits will be overridden by incoming changes. Please request help in #tech-support to resolve conflicts, and please do not run StartProject.bat until the issue is resolved.")
+    elif "unmerged files" in out or "merge_head exists" in out:
+        # we can't abort anything, but don't let stash linger to restore the original repo state
+        pop_if_stashed()
+        error_state("You are in the middle of a merge. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.", fatal_error=True)
+    elif "unborn" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Unborn branch detected. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count)
+        else:
+            handle_error("You are on an unborn branch. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
+    elif "no remote" in out or "no such remote" in out or "refspecs without repo" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Remote repository not found. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count, 2)
+        else:
+            handle_error("The remote repository could not be found. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
+    elif "cannot open" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Git file info could not be read. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count, 3)
+        else:
+            handle_error("Git file info could not be read. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
     else:
-        pblog.error("Aborting the rebase, an unknown error occured. Request help on #tech-support to resolve conflicts, and please do not run StartProject.bat until issue is solved.")
-        pbgit.abort_rebase()
-        pbgit.stash_pop()
-        error_state(True)
+        # We have no idea what the state of the repo is. Do nothing except bail.
+        error_state("Aborting the repo update because of an unknown error. Request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.", fatal_error=True)
+
+    if os.name == "nt":
+        subprocess.Popen("git lfs prune -c ; git lfs dedup", shell=True, creationflags=subprocess.DETACHED_PROCESS, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
https://www.youtube.com/watch?v=f1E_stKKNss

As seen in the video, it's still somehow blocking the main execution

Reply:
Should be fixed now

Reply:
Problem still persists

Reply:
For sake of this PR, I recommend including that change in another pull request.

Reply:
In another comment, you said that the operation still took 28 seconds, when commenting out this. Is it really blocking if it's not adding anything to the execution time? What's going on?

Reply:
With pruning enabled, it takes more then 28 seconds, around ~35-40 probably.

Reply:
Should be fixed, I was able to see Unreal splash screen while prune was still going.

Reply:
The problem still persists on my end, but after the recent discussion, we will push it as is into promoted.